glib/src/lib.rs

```diff
@@ -131,8 +131,10 @@ pub mod closure;
 mod convert;
 pub use self::convert::*;
 mod enums;
-mod functions;
-pub use self::functions::*;
+// We need this to prevent shadowing `auto::functions` reexport from above.
+#[path = "functions.rs"]
+mod inner_functions;
+pub use self::inner_functions::*;
```

Comment:
I would rather make the change to do `pub(crate) auto_functions;` in the auto generated code and we re-export it content from a functions module.

Reply:
If I understand what you said correctly, it would still be the same issue: two `functions` at the top level.

Reply:
Or we could rename the manual one to `manual_functions.rs`? However, what is even the clippy warning here and why don't we get the same for all the other cases where we have a module in `auto` and at the top-level with the same name?

Reply:
> However, what is even the clippy warning here and why don't we get the same for all the other cases where we have a module in auto and at the top-level with the same name?

Because we do `pub mod functions` in both, which I believe is wrong. We should use a `pub (crate)` and re-export these auto functions from the manual functions, similar how we handle prelude & builders.

Reply:
Yes

Reply:
`pub(crate)` won't solve the problem that rustc doesn't know what `functions` is actually. Hence why the rename is needed in the first place. I'll rename the file.

Reply:
> `pub(crate)` won't solve the problem that rustc doesn't know what `functions` is actually. Hence why the rename is needed in the first place. I'll rename the file.

Yes, I rather rename the auto file, as that is easier to do then renaming all the other files